render/cachedAttrsIsStaticMap.js

```diff
@@ -0,0 +1,7 @@
+"use strict"
+
+// This map manages the following:
+// - Whether an attrs is cached attrs generated by compileSelector().
+// - Whether the cached attrs is "static", i.e., does not contain any form attributes.
+// These information will be useful to skip updating attrs in render().
+module.exports = new Map()
```

Comment:
Let's not leak memory here.

```suggestion
module.exports = new WeakMap()
```

Alternatively (and I like this idea better), try using a `cachedAttrsIsStatic` symbol instead of a map. Then, you can do the "is cached" check via `cachedAttrsIsStatic in vnode.attrs` and the "may have form attrs" check via `!vnode.attrs[cachedAttrsIsStatic]`. This would prevent memory leaks in the same way weak maps do, and it'd also allow for engines (like V8) to speed up lookups by caching the hash key.

If that change hurts performance, then just use a weak map.